harness/determined/core/_train.py

```diff
@@ -89,10 +112,55 @@ def report_training_metrics(
             data=det.util.json_encode(body),
         )
 
-        # Also sync tensorboard.
-        if self._tbd_writer and self._tbd_mgr:
+        if self._tbd_sync_mode == TensorboardSyncMode.Auto:
+            self.write_training_tensorboard_metrics(steps_completed, metrics, batch_metrics)
+            if self._tbd_mgr:
+                self._tbd_mgr.sync()
+
+    def write_training_tensorboard_metrics(
+        self,
+        steps_completed: int,
+        metrics: Dict[str, Any],
+        batch_metrics: Optional[List[Dict[str, Any]]] = None,
+    ) -> None:
+        if self._tbd_writer:
             self._tbd_writer.on_train_step_end(steps_completed, metrics, batch_metrics)
-            self._tbd_mgr.sync()
+
+    def get_tensorboard_base_path(self) -> Optional[pathlib.Path]:
+        """
+        Get TensorBoard log directory path.
+        """
+        return self._tbd_mgr.base_path if self._tbd_mgr else None
+
+    def _auto_upload_tb_files(self) -> None:
+        """
+        Uploader for use by Trial classes
+        """
+        if self._distributed.rank == 0:
+            self.upload_tensorboard_files(
+                selector=lambda _: True,  # upload metrics and profiling data
+                mangler=get_rank_aware_path,
+            )
+        else:
+            self.upload_tensorboard_files(
+                selector=lambda p: not p.match("*tfevents*"),
+                mangler=get_rank_aware_path,
+            )
```

Comment:
The Core API is consumed by the TrialControllers, and so this should maybe be a helper function on the base TrialController, or maybe a standaline utility function somewhere.  But the Core API should not have helper functions for our training loops which consume the Core API.